### PR TITLE
feat: add form deletion

### PR DESCRIPTION
**Pull request by dodumosu**
this PR adds the ability to delete forms.

see #569

### CODE REVIEW

File: apollo/formsframework/views_forms.py [hunk 1]
```diff
@@ -429,3 +433,14 @@ def import_form_schema():
         form.save()
 
     return redirect(url_for('formsview.index'))
+
+
+@route(bp, '/delete', methods=['POST'])
```
Comment (takinbo) [inline apollo/formsframework/views_forms.py]: @dodumosu all routes are now being handled within `views_admin.py`. Maybe you could use the same format for all the other views there?

**Review by takinbo — CHANGES_REQUESTED** [review]

**Comment by takinbo** [comment]
PR works most of the time but occasionally, this happens:

screenshot-1572872817

**Review by takinbo — CHANGES_REQUESTED** [review]
PR needs some bug fixing to be merged.

File: apollo/templates/admin/form_list.html [hunk 2]
```diff
@@ -177,6 +189,21 @@ <h5 id="formImportModalLabel" class="modal-title">{{ _('Import Form') }}</h5>
...
+  var importSubmitButton = document.getElementById('btnImportSubmit');
+  importSubmitButton.addEventListener('click', function() {
+    importSubmitButton.disabled = true;
+  });
+  var formDeleteConfirm = function(event) {
+    // for some reason, the event handler activates on clicking the icon,
+    // and the target is, well, the icon. so we have to go up one more
+    // level if that's the case
+    var form = event.target.localName.toLowerCase() === 'button' ? event.target.parentElement : event.target.parentElement.parentElement;
+    var result = confirm("{{ _('Are you sure you want to delete this form? All submissions for it will be deleted too!') }}");
```
Comment (takinbo) [inline apollo/templates/admin/form_list.html]: Can we rephrase to `"All data linked to this form will be deleted too."`?

**Review by takinbo — CHANGES_REQUESTED** [review]

**Review by takinbo — CHANGES_REQUESTED** [review]
I'm still unable to delete forms. See this [video](https://youtu.be/MPCclS4HJuM) for a demonstration.

**Comment by dodumosu** [comment]
was the worker process running when you used it? the actual deletion is done in a worker thread, because it deletes all submissions first.

**Comment by takinbo** [comment]
> was the worker process running when you used it? the actual deletion is done in a worker thread, because it deletes all submissions first.

Yes the worker thread was running and there were no submissions.

**Comment by dodumosu** [comment]
that's definitely weird, because i tried it out and didn't have any issues: https://vimeo.com/510580040

please would you check if the form is still in the database? unless that record is locked somehow, i don't know what else might cause it to fail.

i may need to move it to run on the web thread, but the issue is if the deletion takes time, it'll take the same time to return to the user.

**Pull request closed without merging**